vispy/visuals/volume.py

```diff
@@ -572,6 +592,26 @@ def texture_is_inverted(self):
         if self._texture_limits is not None:
             return self._texture_limits[1] < self._texture_limits[0]
 
+    def _check_clims_in_texture_range(self, clim):
+        inv_tex_outside_limits = (clim[0] > self._texture_limits[0]) or (clim[1] < self._texture_limits[1])
+        tex_outside_limits = (clim[0] < self._texture_limits[0]) or (clim[1] > self._texture_limits[1])
+        return self.texture_is_inverted and inv_tex_outside_limits or tex_outside_limits
+
+    def _check_clims_in_threshold_range(self, clim):
+        # if the clim range is too small of a percentage of the last-used texture range,
+        # posterization may be visible, so downscale the texture range.
+        range_ratio = np.subtract(*clim) / abs(np.subtract(*self._texture_limits))
+        return np.abs(range_ratio) < self._clim_range_threshold
+
+    def _check_clim_requires_rescale(self, clim):
+        if self._texture_float:
+            return False
+        if self._check_clims_in_texture_range(clim):
+            return True
+        if self._check_clims_in_threshold_range(clim):
+            return True
+        return False
+
     @clim.setter
```

Comment:
looks like these got split in a previous PR... but it might be nice to bring `@clim.setter` back together with the original `@property` getter declaration

Reply:
You mean that visually the two parts of the `clim` property are not together? Yeah I can move that.

Reply:
yep... some linters will actually yell at you for that as well (for instance, flake8 is telling me this is a redefinition, unless I put them together)